systemd/Dockerfile.vendor.sccache-dist

```diff
@@ -0,0 +1,21 @@
+# A version of `Dockerfile.build.sccache-dist` that skips the build stage and
+# instead copies over the locally pre-built `sccache-dist` binary.
+# TODO: Switch to BuildKit with Docker/Compose support when available, which
+# allows us to unify these into a single Dockerfile
```

Comment:
My Docker-fu is not that great and maybe there's a way to share a single Dockerfile that allows to either vendor a binary from outside or build the project if needed, without resorting to Buildkit (IIUC Compose needs a very recent version that even supports that). @drahnr @gww-parity do you have an idea if we can deduplicate `Dockerfile.{build,vendor}.sccache-dist` somehow?

Reply:
The vendor one is just to save time in the CI while the build one is to provide a more self-contained example Dockerfile that can be used by others (technically vendor can be as well but the user needs to be on a compatible distro or know to compile targeting `musl`)